### PR TITLE
HPCC-22490 CStreamLineReader returns eof even if data returned

### PR DESCRIPTION
**Pull request by shamser**
Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by shamser** [comment]
@ghalliday Please can  you review

File: system/jlib/jio.cpp [hunk 1]
```diff
@@ -1387,42 +1387,23 @@ class CStreamLineReader : public CSimpleInterfaceOf<IStreamLineReader>
...
-                    }
-                    else // must output what we have and read 1 more byte to check for \n
-                    {
-                        out.append(currentPtr-startPtr-(preserveEols?0:1), startPtr);
-                        if (!refill())
-                            return true;
-                        else if ('\n' != *currentPtr)
-                            return false;
-                        ++currentPtr;
+                    out.append(currentPtr-startPtr-(preserveEols?0:1), startPtr);
```
Comment (ghalliday) [inline system/jlib/jio.cpp]: this change introduces a subtle change - lines that terminate in \r\n will write all but 1 byte, and then append the \n.  That *might* cause the target buffer to reallocate.  I doubt it is significant though.

**Review by ghalliday — COMMENTED** [review]
@shamser the change looks ok - except I think the unit test in jlibtests also needs updating.  Smoketest should pick it up.

File: system/jlib/jio.cpp (hunk 1 above)
Comment (shamser) [reply system/jlib/jio.cpp]: There doesn't seem to be any need to preserve line terminators in any of the current code that uses CStreamLineReader.  We could set all of them to preserveEols==false? (I can also add a comment to state the point about the in the code about this so someone can modify if the need arises)

**Comment by shamser** [comment]
@ghalliday I've updated the smoketests.   Shall I squash?

**Comment by ghalliday** [comment]
@shamser yes, please squash - I'll merge

**Comment by shamser** [comment]
@ghalliday I've squashed.

**Pull request merged**